aten/src/ATen/native/TensorShape.cpp

```diff
@@ -1150,11 +1150,21 @@ Tensor as_strided_tensorimpl_meta(const Tensor& self, IntArrayRef size, IntArray
   return result;
 }
 
+template <typename T>
+inline void setStridedUnchecked(
+    const Tensor& self,
+    ArrayRef<T> size,
+    ArrayRef<T> stride,
+    T storage_offset) {
+  auto* self_ = self.unsafeGetTensorImpl();
+  self_->set_sizes_and_strides(size, stride, c10::make_optional(storage_offset));
+}
+
 Tensor as_strided_tensorimpl_meta_symint(const Tensor& self, SymIntArrayRef sym_size, SymIntArrayRef sym_stride, optional<c10::SymInt> sym_storage_offset_) {
   auto sym_storage_offset = sym_storage_offset_.value_or(self.sym_storage_offset());
   auto result = at::detail::make_tensor<TensorImpl>(
       c10::TensorImpl::VIEW, Storage(self.storage()), self.key_set(), self.dtype());
-  setStrided(result, sym_size, sym_stride, sym_storage_offset);
+  setStridedUnchecked(result, sym_size, sym_stride, sym_storage_offset);
```

Comment:
as_strided calls are the usual reason we generate a guard on a base symbol. Removing all the tests is probably the wrong thing to do, but we're also generating these guards for stupid reasons, e.g., check out this godforsaken guard

```
File "/data/users/ezyang/a/pytorch/torch/fx/experimental/proxy_tensor.py", line 185, in wrap_with_proxy
    set_meta(proxy, e)
  File "/data/users/ezyang/a/pytorch/torch/fx/experimental/proxy_tensor.py", line 131, in set_meta
    proxy.node.meta['val'] = snapshot_fake(val)
  File "/data/users/ezyang/a/pytorch/torch/fx/experimental/proxy_tensor.py", line 121, in snapshot_fake    return val.detach()
  File "/data/users/ezyang/a/pytorch/torch/_subclasses/fake_tensor.py", line 896, in __torch_dispatch__    r = func(*args, **kwargs)
  File "/data/users/ezyang/a/pytorch/torch/_ops.py", line 285, in __call__
    return self._op(*args, **kwargs or {})
  File "/data/users/ezyang/a/pytorch/torch/_decomp/decompositions.py", line 1572, in nop_decomposition
    return aten.alias(x)
  File "/data/users/ezyang/a/pytorch/torch/_ops.py", line 500, in __call__
    return self._op(*args, **kwargs or {})
  File "/data/users/ezyang/a/pytorch/torch/_meta_registrations.py", line 1258, in meta_alias
    return self.view(self.shape)
  File "/data/users/ezyang/a/pytorch/torch/_refs/__init__.py", line 3935, in view
    return _reshape_view_helper(a, *shape, allow_copy=False)
  File "/data/users/ezyang/a/pytorch/torch/_refs/__init__.py", line 3140, in _reshape_view_helper
    return prims.view_of(a)
  File "/data/users/ezyang/a/pytorch/torch/_ops.py", line 285, in __call__
    return self._op(*args, **kwargs or {})
  File "/data/users/ezyang/a/pytorch/torch/_prims/__init__.py", line 1782, in _view_of_meta
    return a.as_strided(a.shape, a.stride(), a.storage_offset())
```

I can't even... what the fuck lol

Reply:
wtf indeed